modules/rest-util/src/blaze/middleware/fhir/resource.clj

```diff
@@ -109,19 +132,43 @@
                     :http/status 415))
     (if (str/blank? (slurp (:body request)))
       (assoc request :body nil)
-      (ba/incorrect "Content-Type header expected, but is missing."))))
+      (ba/incorrect "Expected Content-Type header for FHIR resources."))))
+
+(defn- binary-resource-request [request]
+  (if-let [content-type (request/content-type request)]
+    (cond
+      (json-request? content-type) (resource-request-json request)
```

Comment:
you can't use that json-request? function, because requesting application/json should return the binary content only _format and application/fhir+json should return the resource